app/policies/observation_policy.rb

```diff
@@ -0,0 +1,9 @@
+class ObservationPolicy
+  include ExceptionPolicy
+
+  def can_destroy!
+    role = Role.new(@user, @record.proposal)
+    check(@user.id == @record.user_id || role.client_admin? || @user.admin?,
+          "Only the observer can delete themself")
```

Comment:
Sure. I started with the most restrictive rule, but I don't see any harm in loosening it up the match creation